docs/modules/default-architecture/pages/generation.adoc

```diff
@@ -0,0 +1,123 @@
+
+We recommend you start with LangChain's basic prompt template and OpenAI for the embedding and querying.
+[source,console]
+----
+template = """
+Answer the question based only on the supplied context. \
+If you don't know the answer, say you don't know the answer.
+Context: {context}
+Question: {question}
+Your answer:
+"""
+----
+
+== Further prompt refinement
+
+This is a good prompt to get started, but you can further refine it to solve your problem and improve the docs experience.
+
+How can you improve the first prompt?
+
+* Users want to know they aren't getting bad information or wasting API calls.
+* Operators want to know the chatbot is functioning correctly.
+* Users who want to know more would like a quick link to supporting documentation.
+* Users have deeper questions, and your engineers can help! Maybe they could connect on Slack or Discord.
+
+You decide to add a few things to the prompt template:
+[source,console]
+----
+template = """
+You are a tool called IRL Company Chatbot. \
+Your purpose is to use the below documentation from the
+IRL Company to answer the subsequent documentation questions. \
+Also, if possible, give the reference URLs
+according to the following instructions. The way to create the URLs is: \
+add "https://docs.irl.ai/docs/" before the "slug" value of the document. \
+For any URL references that start with "doc:" or "ref:" \
+use its value to create a URL by adding "https://docs.irl.ai/docs/" before that value. \
+For reference URLs about release notes add "https://docs.irl.ai/changelog/" \
+before the "slug" value of the document. \
+Do not use page titles to create urls. \
+If the answer cannot be found in the documentation, write "I could not find an answer. \
+Join our [Slack community](https://www.irl.ai/slackinvite) for further clarifications." \
+Do not make up an answer or give an answer that does not exist in the provided context. \
+Context: {context}
+Question: {question}
+Helpful answer:"""
+----
+
+This solves your problem and improves the docs experience.
+
+* Users know they're getting good information, because the sources are clearly identified. Operators can see the bot is functioning correctly.
+* Users who want to know more can click the reference links. The chatbot knows how to build links to your docs without manual intervention.
+* Users who have deeper questions can connect with your engineers on Slack.
+
+The additional prompting defines context and new objectives for the LLM. However, just like humans, LLMs can get confused if the prompt is poorly organized or contains non-relevant text. We'll further refine the prompt with the CO-STAR framework.
+
+== CO-STAR prompt framework
+
+A simple way to further improve quality of generation is to follow the https://towardsdatascience.com/how-i-won-singapores-gpt-4-prompt-engineering-competition-34c195a93d41#10b2[CO-STAR framework]{external-link-icon}, a template that sections out key aspects that most influence response generation.
```

Comment:
Yeah I wasn't sure if we wanted to link to this random article, but I couldn't find any other good papers dedicated to it. Here's a shorter one that's more focused: https://generativeai.pub/boost-your-gpts-potential-with-co-star-framework-prompt-e53ed0917df0 or https://medium.com/@barrytwice/harnessing-the-power-of-large-language-models-strategies-and-techniques-47229aa6b101

You could also probably just say "the CO-STAR framework, originally developed by GovTech Singapore's Data Science & AI team" without linking